### PR TITLE
[Fix] Fix the display order of layers when the 'Group as layer' option is used

### PR DESCRIPTION
**Pull request by mind84**
Ticket : #4046

Funded by Faunalia

### CODE REVIEW

**Review by rldhont — CHANGES_REQUESTED** [review]
@mind84 can you add js-unit tests ?

**Comment by mind84** [comment]
Hi @rldhont,
> @mind84 can you add js-unit tests ?

I added a simple equality comparison between the output of the new function and the expected number of layers.
If you have any suggestions for improving this, please let me know

Thanks

File: tests/js-units/node/state/maplayer.test.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { LayerIconSymbology, LayerSymbolsSymbology, SymbolIconSymbology } from '
 import { LayerGroupState, LayerVectorState, LayersAndGroupsCollection } from '../../../../assets/src/modules/state/Layer.js';
 
 import { MapLayerLoadStatus, MapGroupState, MapLayerState } from '../../../../assets/src/modules/state/MapLayer.js';
+import exp from 'constants';
```
Comment (rldhont) [inline tests/js-units/node/state/maplayer.test.js:13]: Seems not be used

**Review by rldhont — APPROVED** [review]
Thanks for the test.
It seems 1 line has been added and not used.

File: tests/js-units/node/state/maplayer.test.js (hunk 1 above)
Comment (mind84) [reply tests/js-units/node/state/maplayer.test.js]: Ouch, I didn't notice, sorry. It seems to have been added automatically by my text editor.

**Comment by mind84** [comment]
closes #4046

**Pull request merged**